packages/tools/fetch-tool/src/fluidFetchMessages.ts

```diff
@@ -140,16 +194,28 @@ async function* saveOps(
     // Split into 100K ops
     const chunk = 100 * 1000;
 
+    let sequencedMessages: ISequencedDocumentMessage[] = [];
+    // current sequence number to start writing from
+    let curr = 1;
     // Figure out first file we want to write to
     let index = 0;
     if (files.length !== 0) {
-        index = (files.length - 1);
+        index = files.length - 1;
+        const name = filenameFromIndex(index);
+        const fileContent = fs.readFileSync(`${dir}/messages${name}.json`, { encoding: "utf-8" });
+        const messages: ISequencedDocumentMessage[] = JSON.parse(fileContent);
+        // set curr to be the number of the first op in the last file and add one.
+        curr = messages[0].sequenceNumber + 1;
     }
-    let curr = index * chunk + 1;
 
-    let sequencedMessages: ISequencedDocumentMessage[] = [];
     while (true) {
         const result: IteratorResult<ISequencedDocumentMessage[]> = await gen.next();
+        // if there was gap in ops, change curr to currSeq,
+        // and index should be 0 anyways because we ask user to delete files
+        curr = seqNumMismatch ? currSeq : curr;
```

Comment:
curr should be just curr = messages[0].sequenceNumber and not currseq otherwise you will end up storing more than 100k ops in the last file. Same example: You had 350K ops stored in 4 files and then firstAvailableDelta was 380K after that. seqNumMismatch would be true and you will set curr to 380k and then it would start writing ops from 380k in the last file and and then last file would have 50K plus 100K = 150k ops. So no need for currSeq here.

Reply:
if curr is 350k , will it be able to skip 30k ops and start writing at 380k?

Reply:
and we need to change the 2 asserts in line 230 and 232

Reply:
Read my other comment on asserts. And yes it will skip the 30k ops because the  "gen" stream of ops will not return it. Again here we are just fetching ops from the stream and not deciding what ops are we fetching. You can also test this with a document and understand more about it.

Reply:
i have tested with multiple documents